woocommerce/woocommerce
# My subscriptions action modals

### Changes proposed in this Pull Request:

This PR adds subscription actions. Depending on the subscription state, the update button will trigger a different modal (subscribe, renew, or connect).
Action buttons will now load in the actions column for each subscription and depending on the subscription state they can be install, subscribe, connect, or renew.

### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. [Connect your site to WCCOM](http://localhost:8888/wp-admin/admin.php?page=wc-admin&tab=my-subscriptions&path=%2Fextensions)
2. Confirm that the install button works
3. Click on the renew button and confirm it opens checkout on WCCOM with your renewal added to the cart (you will need an expired subscription for this)
4. Manually install an outdated version of a product for which you don't have a subscription
5. Click on the update button and confirm it asks you to subscribe first
6. Click on the subscribe button and confirm it opens cart on WCCOM with your product added to the cart
7. After you complete checkout [go back to my subscriptions ](http://localhost:8888/wp-admin/admin.php?page=wc-admin&tab=my-subscriptions&path=%2Fextensions)
8. Click connect and confirm that the subscription is connected



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

Add my subscription action buttons and modals.

</details>

![Screenshot 2023-10-24 at 11 14 49](https://github.com/woocommerce/woocommerce/assets/1199991/055c8519-7498-4347-9567-8b995f6611cc)
